src/editor/extended_python_generator.ts

```diff
@@ -39,6 +39,44 @@ export class ExtendedPythonGenerator extends PythonGenerator {
     super('Python');
   }
 
+  init(workspace: Blockly.Workspace){
+    super.init(workspace);
+    // This will have all variables in the defintion 'variables' so we will need to destroy it and make our own
+    delete this.definitions_['variables'];
+
+    const defvars = [];
+    // Add developer variables (not created or named by the user).
+    const devVarList = Blockly.Variables.allDeveloperVariables(workspace);
+    for (let i = 0; i < devVarList.length; i++) {
+      defvars.push(
+        this.nameDB_!.getName(devVarList[i], Blockly.Names.DEVELOPER_VARIABLE_TYPE) + ' = None',
+      );
+    }
+    this.definitions_['variables'] = defvars.join('\n');     
+    // user variables are dealt with in init code generation
+  }
+
+  defineClassVariables(workspace: Blockly.Workspace) : string{
```

Comment:
Can you add a comment stating that this is called from the python generator for the mrc_class_method_def block for the __init__ method?